plugins/lang-plan-a/src/main/java/org/elasticsearch/plan/a/Analyzer.java

```diff
@@ -500,9 +503,74 @@ public Void visitReturn(final ReturnContext ctx) {
         visit(exprctx);
         markCast(expremd);
 
-        returnsmd.allExit = true;
-        returnsmd.allReturn = true;
-        returnsmd.anyReturn = true;
+        returnsmd.methodEscape = true;
+        returnsmd.loopEscape = true;
+        returnsmd.allLast = true;
+
+        returnsmd.count = 1;
+
+        return null;
+    }
+
+    @Override
+    public Void visitTry(final TryContext ctx) {
+        final StatementMetadata trysmd = adapter.getStatementMetadata(ctx);
+
+        final BlockContext blockctx = ctx.block();
+        final StatementMetadata blocksmd = adapter.createStatementMetadata(blockctx);
+        blocksmd.lastSource = trysmd.lastSource;
```

Comment:
`blocksmd.copyContext(trysmd);`? I know you use "context" to mean something and this might not be the right use of that word though.